### DISCUSSION TITLE
Unwaked turbine power outside of optimization?

### DISCUSSION THREAD

**Discussion by mr483** [NatLabRockies/floris#361] [Q&A] [answered]
Calling fi.get_farm_power(no_wake=True) still gives the waked farm power results. If I run fi.get_farm_power(no_wake=True) without having previously called calculate_wake(), the velocities array is empty and the power() function fails to run. 

I see that there is a related issue (Issue https://github.com/NREL/floris/issues/271.), however this solution seems to only apply when working with the yaw optimizer on a wind rose. Currently, I am doing AEP calculations for standard situations (non-steering) and simply calling calculate_wake() followed by get_farm_power() and get_turbine_powers(). 

Is there an easy way I can obtain non-waked turbine-level powers in order to isolate my wake losses? Thanks!

Edit: 
Temporary solution seems to be to call: 
```
fi.floris.flow_field.initialize_velocity_field(fi.floris.grid)
fi.floris.farm.initialize(fi.floris.grid.sorted_indices)  
farm_power_array = fi.get_farm_power() / 1E6 # in MW
turbine_powers = fi.get_turbine_powers() / 1E6 # In MW
```

**Comment by Bartdoekemeijer** [comment]
Hi @mr483. the `no_wake` feature has not yet been implemented in v3.0. We are currently working on reintroducing it. The issue you reference, #271, was still concerned with FLORIS v2.4. The no_wake feature is likely being implemented as part of #369. Will get back to you once the `no_wake` feature it is in there!

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi @mr483. I just created #373. Please check that out and let me know if it works for your needs!

**Reply by mr483** [reply]
I tested your changes locally and confirmed the outputs match my temporary solution, and also match my basic external AEP calculation, so it seems it is working well. Thanks for the fix!
